### PR TITLE
docs: ADR for Customizable Slashing and Jailing

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct `docs:` prefix in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary 
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)
- [ ] Checked that the documentation website can be built and deployed successfully (run `make build-docs`)

## Summary by CodeRabbit

- **New Features**
  - Introduced a new document outlining "Proof of Reputation Consumer Chains" to enhance blockchain security and reduce operational costs.
  - Proposed a shift from traditional slashing penalties to a reputational system for validators, promoting safer redelegation for delegators.
  - Added a document on "Customizable Slashing and Jailing," enabling consumer chains to tailor slashing and jailing conditions based on validator infractions.

- **Documentation**
  - Provided detailed implementation steps and operational cost management strategies for validators under the new frameworks.

### CODE REVIEW

**Comment by jtremback** [comment]
I'll just comment on the most important part here:

```
### Implementation

The implementation of this feature involves the following steps:

- Add a field to `ConsumerAdditionProposal` to enable consumer chains to launch as PoR chains.
- Disable opt in to PoR consumer chains for validators outside of the provider's active set.
- Cryptographic equivocation evidence received for PoR chains results in the misbehaving validators only being tombstoned and not slashed.
- (Optional) PoR consumer chains can switch to PoS chains, but the transition takes at least unbonding period to allow for validators to opt out.  
```

I think that too much theory is being bundled into this proposal and this implementation plan reflects that. Let's just add few new settings that consumer chain owners can set and modify as they wish.

```
slash_for_equivocation: bool
tombstone_for_equivocation: bool
slash_for_downtime: bool
jail_for_downtime: bool
allow_inactive_validators: bool
```

I think the inactive validators ADR probably already has `allow_inactive_validators` or whatever.

```
- (Optional) PoR consumer chains can switch to PoS chains, but the transition takes at least unbonding period to allow for validators to opt out.
```

This is a good one. @sainoe is working on a mechanism like this for fault resolutions. We should get on a call to discuss whether it should be applied more broadly to other consumer chain settings.

**Comment by mpoke** [comment]
> slash_for_equivocation: bool
> tombstone_for_equivocation: bool
> slash_for_downtime: bool
> jail_for_downtime: bool
> allow_inactive_validators: bool

@jtremback `allow_inactive_validators` is indeed part of the power shaping config as a result of the inactive validators feature

I agree that we could do as you suggest for the rest to allow for more general configurations. I would then make the slashing ones to be slashing ratios instead of bools. For example, a consumer can have `slash_for_equivocation` set to 1%. For starters, we can bound these values to the slashing ratios on the Hub (i.e., `slash_for_equivocation` for any consumer cannot be larger than 5%). 

To improve UX, we can set some defaults in Forge:

- PoS consumer chain
```
slash_for_equivocation: 5%
tombstone_for_equivocation: true
slash_for_downtime: 0%
jail_for_downtime: true
```

- PoR consumer chain
```
slash_for_equivocation: 0%
tombstone_for_equivocation: true
slash_for_downtime: 0%
jail_for_downtime: true
```

- testnet chains 
```
slash_for_equivocation: 0%
tombstone_for_equivocation: false
slash_for_downtime: 0%
jail_for_downtime: false
```

**Comment by mpoke** [comment]
@jtremback I updated the ADR to allow consumer chain owners to set their own slashing and jailing params.

File: docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+Proposed
+
+## Context
+
+Interchain Security (ICS) is a cross-chain staking protocol -- it uses the stake on the provider chain as collateral for the Proof of Stake (PoS) on the consumer chains. 
+This means that the voting power of validators validating (aka producing blocks) on the consumer chains is a function of their stake on the provider.
+Moreover, if these validators misbehave on the consumer chains, they get punished on the provider chain. 
+ICS is currently differentiating between two types of infractions -- equivocation and downtime. 
+Depending on the infraction type, the misbehaving validators might be jailed (i.e., removed from the provider validator set) and / or slashed (i.e., a portion of their stake on the provider is being burned).
+For example, validators double sining on consumer chains get slashed and are permanently jailed,
```
Comment (insumity) [inline docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md:22]:
```suggestion
For example, validators double signing on consumer chains get slashed and are permanently jailed, 
```

Comment (insumity) [reply docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md:22]: "Permanently jailed" means tombstoned?

File: docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+These properties enable us to make the following assumption:
+
+* Being permanently removed from the provider validator set is strong enough of a deterrent to misbehaving on consumer chains.
+
+The additional economical security a consumer gets from slashing is limited: 
+Since most of the stake is delegated, slashing punishes delegators more than validators.  
+
+One benefit of slashing is that it acts as a deterrent for someone buying a large amount of staking tokens in order to attack a consumer chain. 
+For example, an attacker could get `$15.000.000` worth of ATOM, which would give them around `1%` voting power on the Cosmos Hub (at the time of this writing).
```
Comment (insumity) [inline docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md:173]:
```suggestion
For example, an attacker could get `$15,000,000` worth of ATOM, which would give them around `1%` voting power on the Cosmos Hub (at the time of this writing).
```

File: docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+* Reduce the economical security of consumer chains with weaker slashing conditions.
+
+#### Economic Security Model without Slashing
+
+The economic security model of most Cosmos chains relies on the following properties:
+
+* validators are not anonymous, which means that they could be legally liable if they are malicious;
+* the delegated PoS mechanism creates a reputation-based network of validators;
+* most validators have most of their stake coming from delegations (i.e., nothing at stake, besides reputation);
+* it is relatively difficult to enter the active validator set and even more so to climb the voting power ladder.
```
Comment (insumity) [inline docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md:163]: What about inactive validators?

File: docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+This enables consumer chains to tradeoff economical security against cost.  
+
+## Decision
+
+To reduce the cost of ICS, consumer chains will be able to customize the slashing and jailing for every type of infraction. 
+As a result, consumer chains can decide on the amount of economic security they want and validators (and their delegators) can decide on the amount of additional risk they want to incur.  
+
+For every consumer chain, we introduce the following slashing and jailing parameters:
+```proto
+message InfractionParameters {
```
Comment (insumity) [inline docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md:61]: Do we also want to parameterize (i.e., enable/disable) [fault resolutions](https://github.com/cosmos/interchain-security/pull/2051/files)?

Comment (mpoke) [reply docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md:61]: Fault resolutions are not implemented yet.

File: docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md (hunk 3 above)
Comment (mpoke) [reply docs/docs/adrs/adr-020-cutomizable_slashing_and_jailing.md:163]: We could disable this for inactive validators. Or even better, users should choose their configuration.

**Review by MSalopek — APPROVED** [review]

**Pull request merged**